pkgs/development/haskell-modules/configuration-common.nix

```diff
@@ -1515,8 +1515,18 @@ self: super: {
   # Upstream issue: https://github.com/haskell-servant/servant-swagger/issues/129
   servant-swagger = dontCheck super.servant-swagger;
 
-  # 2020-11-27: cxx-options is broken in Cabal 3.2.0.0
-  hercules-ci-agent = addSetupDepend super.hercules-ci-agent self.Cabal_3_2_1_0;
+  hercules-ci-agent = super.hercules-ci-agent.override {
+    cachix =
+      # https://github.com/cachix/cachix/pull/361
+      (appendPatch
+        (addBuildDepend super.cachix super.hercules-ci-cnix-store)
+        ./patches/cachix-cnix.patch);
+  };
+
+  hercules-ci-cli = generateOptparseApplicativeCompletion "hci" (
+    # See hercules-ci-optparse-applicative in non-hackage-packages.nix.
+    addBuildDepend (unmarkBroken super.hercules-ci-cli) super.hercules-ci-optparse-applicative
```

Comment:
Two small nitpicks:

- Could you just mark `hercules-ci-cli` as unbroken in `configuration-hackage2nix.nix` rather than using `unmarkBroken` here?  Normally we avoid using `unmarkBroken` in this file.
- It looks like you've added a patch for `hercules-ci-agent`.  Would it be possible to use `fetchpatch` to pull this from upstream instead of adding it to nixpkgs?  Alternatively, aren't you guys the maintainers of cachix?  Could you make a new release of cachix rather than patch it up in nixpkgs?

Other than that, this LGTM.

Reply:
> * Could you just mark `hercules-ci-cli` as unbroken in `configuration-hackage2nix.nix` rather than using `unmarkBroken` here?  Normally we avoid using `unmarkBroken` in this file.

It seems to be marked as broken by the solver, which makes sense because the "hercules-ci-optparse-applicative" dependency isn't on hackage. Hackage2nix doesn't seem to allow me to explicitly mark a package as unbroken, which seems appropriate.

That package is a fork of `optparse-applicative` with a different name to avoid a ton of unnecessary rebuilds. It's not on hackage because I don't want to pollute the package list with a package that should be temporary.

> * It looks like you've added a patch for `hercules-ci-agent`.  Would it be possible to use `fetchpatch` to pull this from upstream instead of adding it to nixpkgs?  Alternatively, aren't you guys the maintainers of cachix?  Could you make a new release of cachix rather than patch it up in nixpkgs?

Hercules CI and Cachix are in different companies nowadays so we can each focus on one project. Normally Domen responds quickly, but this time he's left two PRs sitting for a couple of days, so I assume he's taking a bit of a break. The PR fixes the profiling build along with some tech debt between the two projects.
Meanwhile I do need these changes to build agent 0.8.0 properly and fix the build in Nixpkgs.

Pulling from upstream doesn't seem like a good idea because there's a chance he requests changes that will then make the patch unavailable in Nixpkgs.